aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -492,8 +495,70 @@ void addr_kernel_cuda(TensorIterator &iter, Scalar beta, Scalar alpha) {
   });
 }
 
+// This reduction accumulates results as the type `acc_t`. By default, when
+// `scalar_t` is complex, `acc_t` is the downgraded real number type.
+// Otherwise, `acc_t` and `scalar_t` are the same type.
+template <typename scalar_t, typename acc_t=typename scalar_value_type<scalar_t>::type, typename out_t=typename scalar_value_type<scalar_t>::type>
+void linalg_vector_norm_kernel_cuda_impl(TensorIterator& iter, Scalar val) {
+  double ord_val;
+  if (val.isIntegral(false)) {
+     ord_val = val.to<int64_t>();
+  } else if (val.isFloatingPoint()) {
+     ord_val = val.to<double>();
+  } else {
+     AT_ERROR("linalg_vector_norm_kernel_cuda_impl expects norm to be integer or float");
+  }
+  if (iter.numel() != 0) {
```

Comment:
This might be more readable with the opposite check:

```
if (iter.numel() == 0) {
  iter.output().fill_((ord_val < 0) ? INFINITY : 0);
  return;
}
```

Then the remaining code doesn't need to be nested and the final `else` clause in this function isn't so far separated from its conditional logic.